ql/src/test/org/apache/hadoop/hive/ql/metadata/TestHiveMetaStoreClientApiArgumentsChecker.java

```diff
@@ -146,6 +151,26 @@ public void testGetPartitionsByNames3() throws HiveException {
     hive.getPartitionsByNames(t, new ArrayList<>(), true);
   }
 
+  @Test
+  public void testGetPartitionsByNamesWithSingleBatch() throws HiveException {
+    hive.getPartitionsByNames(t, Arrays.asList("Greece", "Italy"), true);
+  }
+
+  @Test
+  public void testGetPartitionsByNamesWithMultipleEqualSizeBatches()
+      throws HiveException {
+    List<String> names = Arrays.asList("Greece", "Italy", "France", "Spain");
+    hive.getPartitionsByNames(t, names, true);
+  }
+
+  @Test
+  public void testGetPartitionsByNamesWithMultipleUnequalSizeBatches()
+      throws HiveException {
+    List<String> names =
+        Arrays.asList("Greece", "Italy", "France", "Spain", "Hungary");
+    hive.getPartitionsByNames(t, names, true);
```

Comment:
Could you please add some asserts into these test methods like is the result of `getPartitionsByNames` contains partitions we expect.
Example:
```
    List<Partitions> result = objectStore.getAllStoredProcedures(new ListStoredProcedureRequest("hive"));
    Assert.assertEquals(5, result.size());
    assertThat(result.get(0).getName, is("Greece"));
...
```
It would be also good to know how many times was the method `IMetaStoreClient.getPartitionsByNames` called. I think to get this a mock framework should be used. Mockito is already used in some Hive tests.

Reply:
The goal of the tests in this class are explained in the javadoc of `TestHiveMetaStoreClient`. 

> Tests in this class ensure that both tableId and validWriteIdList are sent from HS2

Asserting more things seems to be out of scope for this PR.